daml/ContingentClaims/Claim.daml

```diff
@@ -23,45 +22,45 @@ type F = ClaimF
 -- See [quickstart](../QUICKSTART.md) for detailed explanation.
 -- In Petyon-Jones' paper, this is called 'Contract'.
 -- We renamed it to avoid ambiguity.
--- * `f`, `t` and `x` respectively correspond to the `Observable`, it's input type and the resulting output type.
+-- * `t` and `x` respectively correspond to the `Observation`s, input type and the resulting output type.
 -- * `a` is the representation of an asset, e.g. a `Text` ISIN code.
-data Claim f t x a
+data Claim t x a
   = Zero
       -- ^ Represents an absence of claims.
   | One a
       -- ^ The bearer acquires one unit of `a` *immediately*.
-  | Give (Claim f t x a)
+  | Give (Claim t x a)
       -- ^ The obligations of the bearer and issuer are revesed.
-  | And with lhs: Claim f t x a, rhs: Claim f t x a
+  | And with fst: Claim t x a, snd: Claim t x a, tail: [Claim t x a]
       -- ^ Used to combine multiple rights together.
-  | Or with lhs: Claim f t x a, rhs: Claim f t x a
-      -- ^ Gives the bearer the right to choose between two claims.
-  | Cond with predicate: Inequality f t x a, success: Claim f t x a, failure: Claim f t x a --TODO rename predicate -> `lte` or `inequality`
+  | Or with claims: [Claim t x a]
+      -- ^ Gives the bearer the right to choose between several claims.
```

Comment:
Shouldn't we have at minimum 2 elements here as well if we have it for `And`?

Reply:
Yes. I was testing out the `And` before doing the same for `Or`.

I did have some doubts whether the binary representation should be kept for `Or`, see my comments above on the bermudan option.